[BUG]: UCX job clusters are deployed with incorrect config

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

On AWS, `LEGACY_SINGLE_USER` clusters have passthrough enabled. This results in error when running assessment job
<img width="395" alt="image" src="https://github.com/databrickslabs/ucx/assets/44292934/4bd5f297-ab51-4818-beb2-2667ffa457ce">
```
AnalysisException: org.apache.hadoop.hive.ql.metadata.HiveException: com.amazonaws.SdkClientException: Unable to load AWS credentials from any provider in the chain: [com.databricks.backend.daemon.driver.aws.AwsLocalCredentialContextTokenProvider@4c7592e8: No role specified and no roles available., com.databricks.backend.daemon.driver.aws.ProxiedIAMCredentialProvider@21a98c2: User does not have any IAM roles]
Caused by: HiveException: com.amazonaws.SdkClientException: Unable to load AWS credentials from any provider in the chain: [com.databricks.backend.daemon.driver.aws.AwsLocalCredentialContextTokenProvider@4c7592e8: No role specified and no roles available., com.databricks.backend.daemon.driver.aws.ProxiedIAMCredentialProvider@21a98c2: User does not have any IAM roles]
Caused by: SdkClientException: Unable to load AWS credentials from any provider in the chain: [com.databricks.backend.daemon.driver.aws.AwsLocalCredentialContextTokenProvider@4c7592e8: No role specified and no roles available., com.databricks.backend.daemon.driver.aws.ProxiedIAMCredentialProvider@21a98c2: User does not have any IAM roles]
```

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Cloud

AWS

### Operating System

macOS

### Version

latest via Databricks CLI

### Relevant log output

_No response_

--- a/src/databricks/labs/ucx/installer/workflows.py
+++ b/src/databricks/labs/ucx/installer/workflows.py
@@ -727,7 +727,7 @@ class WorkflowsDeployment(InstallationMixin):
                 jobs.JobCluster(
                     job_cluster_key="main",
                     new_cluster=compute.ClusterSpec(
-                        data_security_mode=compute.DataSecurityMode.LEGACY_SINGLE_USER,
+                        data_security_mode=compute.DataSecurityMode.LEGACY_SINGLE_USER_STANDARD,
                         spark_conf=self._job_cluster_spark_conf("main"),
                         custom_tags={"ResourceClass": "SingleNode"},
                         num_workers=0,

